### PR TITLE
Added commands to manage shell extensions

### PR DESCRIPTION
**Pull request by ngdentdev**
Created register, unregister and list commands to manage shell extensions

### CODE REVIEW

File: src/shell/Commands/ExtensionCommand.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+using McMaster.Extensions.CommandLineUtils;
+
+namespace NeoShell.Commands
+{
+  [Command("extension", Description = "Commands to manage NEO shell extensions")]
```
Comment (devhawk) [inline src/shell/Commands/ExtensionCommand.cs:5]: nitpick - Neo is only capitalized like this when referencing the NEO token.

File: src/shell/Program.cs [hunk 2]
```diff
@@ -14,6 +14,7 @@ namespace NeoShell
     [Subcommand(typeof(ConnectCommand))]
     [Subcommand(typeof(ShowCommand))]
     [Subcommand(typeof(TransferCommand))]
+    [Subcommand(typeof(ExtensionCommand))]
```
Comment (devhawk) [inline src/shell/Program.cs:17]: nitpick alphabetize

File: src/shell/Commands/ExtensionCommand.cs [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  [Subcommand(typeof(Register), typeof(List), typeof(Unregister))]
```
Comment (devhawk) [inline src/shell/Commands/ExtensionCommand.cs:6]: nitpick alphabetize

File: src/shell/Commands/ExtensionCommand.Unregister.cs [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+            internal int OnExecute(CommandLineApplication app, IConsole console)
+            {
+                try
+                {
+                    string rootPath = this.fileSystem.Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.UserProfile), BaseFolder);
+
+                    string extensionsFilePath = this.fileSystem.Path.Combine(rootPath, ExtensionsFile);
+                    string extensionsJsonContent = this.fileSystem.File.ReadAllText(extensionsFilePath);
+                    JArray extensionsArray = JArray.Parse(extensionsJsonContent);
```
Comment (devhawk) [inline src/shell/Commands/ExtensionCommand.Unregister.cs:33]: Should we be mapping the loosely typed JSON to C# types? Not sure it's worth it, but figured we should discuss

**Review by JohndeVadoss — APPROVED** [review]

File: src/shell/Commands/ExtensionCommand.cs [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        private const string ExtensionsFile = "extensions.json";
+
+        internal int OnExecute(CommandLineApplication app, IConsole console)
+        {
+            console.WriteLine("You must specify at a subcommand.");
+            app.ShowHelp(false);
+            return 1;
+        }
+
+        private static JArray LoadExtensions(IFileSystem fileSystem)
```
Comment (devhawk) [inline src/shell/Commands/ExtensionCommand.cs:20]: should also use this from LoadShellExtensions

File: src/shell/Commands/ExtensionCommand.Unregister.cs (hunk 4 above)
Comment (ngdentdev) [reply src/shell/Commands/ExtensionCommand.Unregister.cs]: Refactored the code to use existing ShellExtensions and ShellExtenion classes. Thank you for reminding me those exist.

File: src/shell/Models/ShellExtensions.cs [hunk 6]
```diff
@@ -1,10 +1,14 @@
 using System.Diagnostics.CodeAnalysis;
 using Newtonsoft.Json;
+using System.IO.Abstractions;
 
 namespace NeoShell.Models
 {
     class ShellExtensions : List<ShellExtension>
```
Comment (devhawk) [inline src/shell/Models/ShellExtensions.cs:7]: This should be a static class

File: src/shell/Commands/ExtensionCommand.Register.cs [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+                    return 0;
+                }
+                catch (Exception ex)
+                {
+                    app.WriteException(ex, showInnerExceptions: true);
+                    return 1;
+                }
+            }
+
+            private void CreateBaseFilesIfNotExist()
```
Comment (devhawk) [inline src/shell/Commands/ExtensionCommand.Register.cs:60]: Should this be in ShellExtensions class?

**Review by devhawk — APPROVED** [review]
A couple more minor suggestions to address then I think this is ready to go

File: src/shell/Commands/ExtensionCommand.Register.cs (hunk 7 above)
Comment (ngdentdev) [reply src/shell/Commands/ExtensionCommand.Register.cs]: good catch moved to shell extensions

File: src/shell/Models/ShellExtensions.cs (hunk 6 above)
Comment (ngdentdev) [reply src/shell/Models/ShellExtensions.cs:7]: There are still a few instance methods like Persist, ToJson etc.

**Pull request merged**